src/AWS.Deploy.CLI/Commands/DeployCommand.cs

```diff
@@ -451,7 +454,7 @@ private void ConfigureDeploymentFromConfigFile(Recommendation recommendation, Us
                         throw new InvalidOverrideValueException(DeployToolErrorCode.InvalidValueForOptionSettingItem, $"Invalid value {optionSettingValue} for option setting item {optionSettingJsonPath}");
                     }
 
-                    _optionSettingHandler.SetOptionSettingValue(optionSetting, settingValue);
+                    await _optionSettingHandler.SetOptionSettingValue(optionSetting, settingValue);

```

Comment:
At this point when loading up the recipe I'm not sure we know if this is for new stack or redeployment to an existing stack. If it is redeployment to an existing stack then it should throw an error when new beanstalk app name already exists but I think with this logic it will.